src/mscorlib/shared/System/Number.Parsing.cs

```diff
@@ -913,5 +927,37 @@ private static bool TrailingZeros(ReadOnlySpan<char> s, int index)
         }
 
         private static bool IsWhite(char ch) => ch == 0x20 || (ch >= 0x09 && ch <= 0x0D);
+
+        private static bool NumberBufferToDouble(ref NumberBuffer number, ref double value)
+        {
+            double d = NumberToDouble(ref number);
+            uint e = DoubleHelper.Exponent(d);
+            ulong m = DoubleHelper.Mantissa(d);
+
+            if (e == 0x7FF)
+            {
+                return false;
+            }
+
+            if (e == 0 && m == 0)
+            {
+                d = 0;
+            }
+
+            value = d;
+            return true;
+        }
+
+        private static class DoubleHelper
+        {
+            public static unsafe uint Exponent(double d) =>
+                (*((uint*)&d + 1) >> 20) & 0x000007ff;
+
+            public static unsafe ulong Mantissa(double d) =>
+                *((uint*)&d) | ((ulong)(*((uint*)&d + 1) & 0x000fffff) << 32);
```

Comment:
This looks more complext than it needs to be. Would `*((ulong*)&d) & 0x000fffffffffffff` work better?

Reply:
Definitely.  I'd just copied the code from corert, but I'm not sure why it was done in such a convoluted manner... took me a minute to work through what it was actually doing to convince myself it was the same :)